### PR TITLE
Remove an example code of `SkoptSampler`

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation

Since Scikit-optimize has been inactive and we have identified a compatibility issues with newer NumPy versions.

## Description of the changes

Remove an example code of `SkoptSampler` to make doctest pass again.

### CODE REVIEW

File: optuna/integration/skopt.py [hunk 1]
```diff
@@ -31,24 +31,8 @@
...
-            def objective(trial):
-                x = trial.suggest_float("x", -10, 10)
-                y = trial.suggest_int("y", 0, 10)
-                return x**2 + y
-
-
-            sampler = optuna.integration.SkoptSampler()
-            study = optuna.create_study(sampler=sampler)
-            study.optimize(objective, n_trials=10)
+    The use of SkoptSampler is highly not recommended, as the development of Scikit-Optimize
```
Comment (not522) [inline optuna/integration/skopt.py:34]: How about adding a link?

```suggestion
    The use of :class:`~optuna.integration.SkoptSampler` is highly not recommended, as the development of Scikit-Optimize
```

**Comment by toshihikoyanase** [comment]
@cross32768 Could you review this PR, please?

**Review by cross32768 — APPROVED** [review]
LGTM after above pointing is resolved

File: optuna/integration/skopt.py (hunk 1 above)
Comment (c-bata) [reply optuna/integration/skopt.py:34]: Thank you for your review! I deliberately did not make it a link. Do you think we should add a link even from the same documentation page?

Comment (not522) [reply optuna/integration/skopt.py:34]: I think the decorative notation makes it easier to read. Or, replacing it with "this sampler" is OK.

Comment (c-bata) [reply optuna/integration/skopt.py:34]: Sure. Let me apply your suggestion :pray:

**Review by not522 — APPROVED** [review]
LGTM

**Comment by toshihikoyanase** [comment]
The `doctest` CI of the master branch passed ([log](https://github.com/optuna/optuna/actions/runs/4751719153/jobs/8441181680)). Thank you!

**Pull request merged**